tests/examples/test_chat.py

```diff
@@ -0,0 +1,94 @@
+#!/bin/env python3
+import asyncio
+import sys
+
+import click
+import multiaddr
+import pytest
+
+from libp2p.libp2p import *
+from peer.peerinfo import info_from_p2p_addr
+
+# TODO: change once muxed_connection supports extracting protocol id from messages
```

Comment:
I believe this has been enabled with `protocol_muxer` which is equivalent of `multistream-select`. I will let @stuckinaboot shed more light on this.

Reply:
haaa could be indeed this has been fixed since then. I had this code laying around for a while. I'll check. Thanks

Reply:
So I tested and yes indeed protocol id is now passed during negotiation. So I updated the code and added a test for this too.

Reply:
@zaibon great thanks, if you have any questions about negotiation feel free to let me know.